src/surface.rs

```diff
@@ -121,3 +120,53 @@ pub fn get_outputs(surface: &Proxy<wl_surface::WlSurface>) -> Vec<Proxy<wl_outpu
         .outputs
         .clone()
 }
+
+/// Surface Manager
+#[derive(Clone)]
+pub struct SurfaceManager {
+    output_manager: OutputMgr,
+    surfaces: Arc<Mutex<Vec<Proxy<wl_surface::WlSurface>>>>,
+}
```

Comment:
I think at this point the `SurfaceManager` could also own a `Proxy<WlCompositor>`. This would allow to remove the need for passing it as argument to the `create_surface` method. The `wl_compositor` global is not really something that is supposed to change during the lifetime of the client.

Reply:
The problem is that the SurfaceManager needs to be created before the GlobalManager which is needed to get the WlCompositor.

Reply:
Ah right. The usual chicken and egg problem...

Can't we work around that though? The `OutputMgr` cannot be created outside of the `Environment`, it may be acceptable for the `SurfacesMgr` to be the same way. This would, there could be a tighter coupling between its internals and the `Environment`, no?